cpp/src/io/parquet/page_data.cu

```diff
@@ -1845,8 +1940,663 @@ __global__ void __launch_bounds__(block_size)
+
+    // copy suffixes into string data
+    uint8_t* so_ptr = strings_out + string_off;
+    if (ln_idx < suffix_db->value_count) {
+      memcpy(so_ptr + prefix_len, suffix_data + suffix_off, suffix_len);
+    }
+    __syncwarp();
+
+    // copy prefixes into string data.
+    StringScan(dba, strings_out, last_string, idx, lane_id);
+
+    if (lane_id == 31) {
+      last_string = strings_out + string_off;
+      suffix_data += suffix_off + suffix_len;
+    }
+    __syncwarp();
+  }
+}
+
 /**
- * @brief Kernel for co the column data stored in the pages
+ * @brief Kernel for computing per-page size information for DELTA_BYTE_ARRAY encoded pages.
+ */
+__global__ void __launch_bounds__(64) gpuComputePageStringSizes(
+  PageInfo* pages, device_span<ColumnChunkDesc const> chunks, size_t num_rows, size_t min_row)
+{
+  // using this to get atomicAdd happy
+  using u64 = unsigned long long int;
+  __shared__ __align__(16) delta_byte_array_state_s db_state;
+  __shared__ __align__(8) u64 total_bytes;
+
+  uint32_t const t       = threadIdx.x;
+  uint32_t const lane_id = t & 0x1f;
+  auto* const prefix_db  = &db_state.prefixes;
+  auto* const suffix_db  = &db_state.suffixes;
+  auto* const page       = &pages[blockIdx.x];
+  auto const* const col  = &chunks[page->chunk_idx];
+
+  // initialize strings info
+  if (t == 0) {
+    page->page_string_data  = nullptr;
+    page->page_strings_size = 0;
+  }
+
+  if (page->encoding != Encoding::DELTA_BYTE_ARRAY || page->num_input_values == 0) { return; }
+
+  // check page bounds to see if we need sizes for this page
+  size_t page_start = col->start_row + page->chunk_row;
+  size_t page_end   = page_start + page->num_rows;
+  if (page_start >= min_row + num_rows or page_end < min_row) { return; }
+
+  // get pointers to start and end of encoded page data
+  auto [cur, end] = page_data_bounds(page);
+
+  if (t == 0) {
+    // initialize the prefixes and suffixes blocks
+    auto const* const suffix_start = FindEndOfBlock(prefix_db, cur, end);
+    FindEndOfBlock(suffix_db, suffix_start, end);
+
+    // initialize total_bytes
+    total_bytes = prefix_db->last_value + suffix_db->last_value;
+  }
+  __syncthreads();
+
+  // step through prefixes and suffixes to get total length
+  while (prefix_db->current_value_idx < num_encoded_values(prefix_db, true)) {
+    auto* const db = t < 32 ? prefix_db : suffix_db;
+
+    for (uint32_t i = 0; i < db->values_per_mb; i += 32) {
+      CalcMiniBlockValues(db, lane_id);
+
+      uint32_t const idx = db->current_value_idx + i + lane_id;
+      u64 const my_len   = idx < db->value_count ? db->value[rolling_index(idx)] : 0;
+
+      // get sum for warp
+      using WarpReduce = cub::WarpReduce<u64>;
+      __shared__ typename WarpReduce::TempStorage temp_storage;
+      // note: sum_len will only be valid on thread 0.
+      u64 const sum_len = WarpReduce(temp_storage).Sum(my_len);
+
+      if (lane_id == 0) { atomicAdd(&total_bytes, sum_len); }
+      __syncthreads();
```

Comment:
```suggestion
      __syncwarp();
```
I think this can just be a syncwarp.  The `atomicAdd` takes care of you at the block level.

Alternately, you could just store the per-warp total_bytes and do a single call to `cudf::detail::single_lane_block_sum_reduce` at the very end to sum them together, with absolute thread value t==0 doing the final store.  Would dodge the atomics entirely.